rules/do-you-use-logging-fakes/rule.md

```diff
@@ -0,0 +1,142 @@
+---
+seoDescription: Learn how to use Microsoft's `FakeLogger` in unit tests to verify logging behavior in .NET applications. This guide shows you how to capture log entries, check log levels, and ensure your logging works as expected during testing.
+type: rule
+title: Do you use Logging Fakes?
+uri: do-you-use-logging-fakes
+authors:
+  - title: Tom Iwainski
+    url: https://www.ssw.com.au/people/thomas-iwainski
+related: []
+created: 2024-11-05T12:52:08.000Z
+archivedreason: null
+guid: dc76970f-deea-4e92-9d55-731e83bee7b5
+---
+
+Unit tests are an essential part of ensuring that your .NET applications work correctly. However, when it comes to testing logging, things can get tricky. You don't want to rely on actual logging infrastructure like file writes or external logging systems during tests, as this could lead to unnecessary complexity and slow down your test suite.
+
+Microsoft provides a simple solution with `FakeLogger`. This logger allows you to capture and verify logging calls in unit tests without needing to integrate with a real logging framework.
+
+By incorporating `FakeLogger` into your tests, you can ensure that your logging functionality is correct and meets your application's needs, all while maintaining fast and isolated tests.
+
+<!--endintro-->
+
+## What is the FakeLogger?
+
+The FakeLogger is a mock logger provided by Microsoft for unit testing. It is part of the Microsoft.Extensions.Logging namespace and is used primarily in tests to verify that logging actions occur during the execution of the application, without sending log output to external destinations like the console or log files. Instead, it "fakes" the log output, allowing assertions to be made on the log entries.
+
+The advantage of using `FakeLogger` is that it behaves like a real `ILogger<T>`, but it stores the logs in memory, making them available for examination during your tests.
+
+## Benefits of using FakeLogger
+
+- **No External Dependencies**: Since `FakeLogger` does not actually write logs anywhere (such as to disk or a remote server), it keeps your unit tests isolated and fast.
```

Comment:
```suggestion
- **No External Dependencies** - Since `FakeLogger` does not actually write logs anywhere (such as to disk or a remote server), it keeps unit tests isolated and fast
```